lib/WUI/nhttp/file_info.cpp

```diff
@@ -33,6 +35,75 @@ namespace nhttp::printer {
 using namespace handler;
 
 JsonResult FileInfo::DirRenderer::renderState(size_t resume_point, JsonOutput &output, FileInfo::DirState &state) const {
+    if (api == APIVersion::Octoprint) {
+        return renderStateOctoprint(resume_point, output, state);
+    } else {
+        return renderStateV1(resume_point, output, state);
+    }
+}
+
+JsonResult FileInfo::DirRenderer::renderStateV1(size_t resume_point, JsonOutput &output, FileInfo::DirState &state) const {
+    struct stat st {};
+
+    // Keep the indentation of the JSON in here!
+    // clang-format off
+    JSON_START;
+    JSON_OBJ_START;
+
+        JSON_CONTROL("\"type\":\"FOLDER\"") JSON_COMMA;
+        JSON_FIELD_BOOL("ro", false) JSON_COMMA;
+        // Note: we can only get a timestamp for folders that are not
+        // /usb/ because that is the root of FATFS and stat() don't work on it
+        if (stat(state.filepath, &st) == 0) {
+            // Note: We need the timestamp to be persistent betweeen resumes,
+            // so we save it to state.
+            state.base_folder_timestamp = st.st_mtime;
```

Comment:
Well, this is one solution. But that seems to be the more complicated (and larger to store long-term) solution.

I was simply thinking of putting the `stat` before the JSON_START… but that would complicate the situation of the `if(stat(...))` :thinking: … which would require a JSON_FIELD_INT_G… hmm, maybe we can leave it as it is.